storage/aptosdb/src/fast_sync_aptos_db.rs

```diff
@@ -0,0 +1,563 @@
+    fn get_state_snapshot_receiver(
+        &self,
+        version: Version,
+        expected_root_hash: HashValue,
+    ) -> Result<Box<dyn StateSnapshotReceiver<StateKey, StateValue>>> {
+        let mut status = self
+            .fast_sync_status
+            .write()
+            .expect("Failed to get write lock of fast sync status");
+        *status = FastSyncStatus::STARTED;
+        self.get_aptos_db_write_ref()
+            .get_state_snapshot_receiver(version, expected_root_hash)
+    }
+
+    fn finalize_state_snapshot(
+        &self,
+        version: Version,
+        output_with_proof: TransactionOutputListWithProof,
+        ledger_infos: &[LedgerInfoWithSignatures],
+    ) -> Result<()> {
+        let status = self.get_fast_sync_status()?;
+        assert_eq!(status, FastSyncStatus::STARTED);
+        self.get_aptos_db_write_ref().finalize_state_snapshot(
+            version,
+            output_with_proof,
+            ledger_infos,
+        )?;
+        let mut status = self
+            .fast_sync_status
+            .write()
+            .expect("Failed to get write lock of fast sync status");
+        *status = FastSyncStatus::FINISHED;
+        Ok(())
+    }
+
+    fn save_transactions(
+        &self,
+        txns_to_commit: &[TransactionToCommit],
+        first_version: Version,
+        base_state_version: Option<Version>,
+        ledger_info_with_sigs: Option<&LedgerInfoWithSignatures>,
+        sync_commit: bool,
+        latest_in_memory_state: StateDelta,
+    ) -> Result<()> {
+        self.get_aptos_db_write_ref().save_transactions(
+            txns_to_commit,
+            first_version,
+            base_state_version,
+            ledger_info_with_sigs,
+            sync_commit,
+            latest_in_memory_state,
+        )
+    }
+
+    fn save_transaction_block(
+        &self,
+        txns_to_commit: &[Arc<TransactionToCommit>],
+        first_version: Version,
+        base_state_version: Option<Version>,
+        ledger_info_with_sigs: Option<&LedgerInfoWithSignatures>,
+        sync_commit: bool,
+        latest_in_memory_state: StateDelta,
+        block_state_updates: ShardedStateUpdates,
+        sharded_state_cache: &ShardedStateCache,
+    ) -> Result<()> {
+        self.get_aptos_db_write_ref().save_transaction_block(
+            txns_to_commit,
+            first_version,
+            base_state_version,
+            ledger_info_with_sigs,
+            sync_commit,
+            latest_in_memory_state,
+            block_state_updates,
+            sharded_state_cache,
+        )
+    }
+}
+
+macro_rules! aptos_db_read_fn {
+    ($(fn $name:ident(&self $(, $arg: ident : $ty: ty $(,)?)*) -> $return_type:ty,)+) => {
+        $(fn $name(&self, $($arg: $ty),*) -> $return_type {
+            self.get_aptos_db_read_ref().$name($($arg),*)
+        })+
+    };
+}
+
+impl DbReader for FastSyncStorageWrapper {
+    aptos_db_read_fn!(
+        fn get_transactions(
```

Comment:
nit: any chance we could still have empty lines between the functions to make it more readable? 😄